kong/api/crud_helpers.lua

```diff
@@ -75,7 +76,18 @@ function _M.paginated_set(self, dao_collection)
   -- This check is required otherwise the response is going to be a
   -- JSON Object and not a JSON array. The reason is because an empty Lua array `{}`
   -- will not be translated as an empty array by cjson, but as an empty object.
-  local result = #rows == 0 and "{\"data\":[],\"total\":0}" or {data = rows, ["next"] = next_url, total = total_count}
+  -- TODO: fixme once https://github.com/openresty/lua-cjson/pull/6 lands in a release
+  local result
+  if #rows == 0 then
+    result = [[{"data":[],"total":0}]]
+  else
+    result = {
+      data = rows,
+      total = total_count,
+      offset = offset,
+      ["next"] = next_url,
+    }
+  end
```

Comment:
It would create a table, which is immediately disposed of again. Current code is more performant.